packages/compiler-core/src/parse.ts

```diff
@@ -239,7 +239,7 @@ function parseChildren(
       }
       // also remove comment nodes in prod by default
       if (
-        !__DEV__ &&
+        (__GLOBAL__ || __ESM_BROWSER__ || !__DEV__) &&
```

Comment:
I wonder if we shouldn't just remove the check of `!__DEV__` and explicitly look for `context.options.comments === false`